Mathlib/Analysis/InnerProductSpace/Dual.lean

```diff
@@ -69,6 +70,61 @@ variable {E}
 theorem toDualMap_apply {x y : E} : toDualMap 𝕜 E x y = ⟪x, y⟫ :=
   rfl
 
+section NullSubmodule
+
+open SeparationQuotient LinearMap
+
+variable (E)
+
+/-- The null space with respect to the norm. -/
+def nullSubmodule : Submodule 𝕜 E where
+  __ := nullSubgroup
+  smul_mem' c x (hx : ‖x‖ = 0) := show ‖c • x‖ = 0 from
+    le_antisymm (norm_smul_le _ _ |>.trans <| by rw [hx, mul_zero]) (norm_nonneg _)
+
+@[simp]
+lemma mem_nullSubmodule_iff {x : E} : x ∈ nullSubmodule 𝕜 E ↔ ‖x‖ = 0 := Iff.rfl
+
+lemma inner_eq_zero_of_left (x y : E) (h : ‖x‖ = 0) :
+    ⟪x, y⟫_𝕜 = 0 := by
+  rw [← norm_eq_zero, ← sq_eq_zero_iff]
+  apply le_antisymm _ (sq_nonneg _)
+  rw [sq]
+  nth_rw 2 [← RCLike.norm_conj]
+  rw [_root_.inner_conj_symm]
+  calc ‖⟪x, y⟫_𝕜‖ * ‖⟪y, x⟫_𝕜‖ ≤ re ⟪x, x⟫_𝕜 * re ⟪y, y⟫_𝕜 := inner_mul_inner_self_le _ _
+  _ = (‖x‖ * ‖x‖) * re ⟪y, y⟫_𝕜 := by rw [inner_self_eq_norm_mul_norm x]
+  _ = (0 * 0) * re ⟪y, y⟫_𝕜 := by rw [(mem_nullSubmodule_iff 𝕜 E).mp h]
+  _ = 0 := by ring
+
+lemma inner_nullSubmodule_right_eq_zero (x y : E) (h : ‖y‖ = 0) : ⟪x, y⟫_𝕜 = 0 := by
+  rw [inner_eq_zero_symm]
+  exact inner_eq_zero_of_left 𝕜 E y x h
```

Comment:
Name doesn't match, it should be `inner_eq_zero_of_right`. (or maybe both of these could just delete the word `of`)
```suggestion
lemma inner_eq_zero_of_right (x : E) {y : E} (h : ‖y‖ = 0) : ⟪x, y⟫_𝕜 = 0 := by
  rw [inner_eq_zero_symm, inner_eq_zero_of_left _ _ h]
```